### PR TITLE
igl | opengl | Fix memory leak in RenderPipelineState::activeAttributesLocations_

### PR DESCRIPTION
**Pull request by vinsentli**
When a RenderCommandEncoder sequentially binds two IRenderPipelineStates, if both of these states have the same VertexInputState, the previous IRenderPipelineState is unable to execute unbindVertexAttributes(). As a result, the activeAttributesLocations_ of the previous RenderPipelineState remains in a continuous state of memory growth and cannot be released.

### CODE REVIEW

**Comment by EricGriffith** [comment]
This PR fixes a memory leak, but it may cause some perf overhead if things are unbound and then immediately rebound. Is there a way to fix the memory leak without the perf overhead?

**Comment by vinsentli** [comment]
@EricGriffith 
My solution is to save the activeAttributesLocations_ of the previous pipeline state, for the new draw call, only disable the attribute locations that are not needed. This way, the number of glEnable/glDisableVertexAttribArray calls can be reduced.

**Comment by vinsentli** [comment]
@corporateshark This PR has opened for a long time ,is there any problem with it ?

**Comment by corporateshark** [comment]
> This PR has opened for a long time ,is there any problem with it ?

@vinsentli Thank you for following up! We’re still conducting some internal testing. Rest assured, as long as this PR remains open, someone on the team will review it at some point.

**Comment by vinsentli** [comment]
@corporateshark Thanks for your reply.

File: src/igl/opengl/RenderCommandAdapter.cpp [hunk 1]
```diff
@@ -355,6 +355,7 @@ void RenderCommandAdapter::willDraw() {
         CLEAR_DIRTY(vertexBuffersDirty_, bufferIndex);
       }
     }
+    pipelineState->unbindPrePipelineVertexAttributes();
```
Comment (corporateshark) [inline src/igl/opengl/RenderCommandAdapter.cpp:358]: unbindPre**v**PipelineVertexAttributes()

**Review by corporateshark — CHANGES_REQUESTED** [review]
Please rebase this on top of the latest `main` branch.

File: src/igl/opengl/RenderCommandAdapter.cpp (hunk 1 above)
Comment (vinsentli) [reply src/igl/opengl/RenderCommandAdapter.cpp]: done

**Comment by vinsentli** [comment]
> Please rebase this on top of the latest `main` branch.

done

**Review by corporateshark — APPROVED** [review]

**Pull request closed without merging**